config/routes.rb

```diff
@@ -56,7 +56,9 @@
 
       json_api_resources :project_roles
 
-      json_api_resources :project_preferences
+      json_api_resources :project_preferences do
+        post "/update_project_settings", to: "project_preferences#update_project_settings", on: :collection
```

Comment:
double check but i think this can be cleaned up and written like the other resource collection routes
``` ruby
collection do 
  post :update_project_settings
end
```